src/types/serialization.js

```diff
@@ -50,12 +50,12 @@ export function serialize(buffer, shift, data, type) {
                 buffer = serialize(buffer, from, fieldData, fieldType.type);
             } else {
                 var end = buffer.length;
-                Uint32(end, buffer, from, from + 4);
+                Uint32(end - shift, buffer, from, from + 4);
```

Comment:
- This requires to update the documentation, doesn't it? Seems like a fundamental change in what segment pointers point to.
- How many unit tests cover this directly? Seems like too little.

Reply:
- Right, this changed should be reflected in documentation. We discussed it with @oudod.
- Now it is only one unit test for each case where segment pointers are used: 1) nested type 2) `String` 2) `FixedBuffer`